lib/services/local/session.go

```diff
@@ -95,3 +97,249 @@ func (s *IdentityService) DeleteAllAppSessions(ctx context.Context) error {
 	}
 	return nil
 }
+
+// GetWebSession returns a web session state for the given user and session id
+func (s *IdentityService) GetWebSession(user, sessionID string) (services.WebSession, error) {
+	return s.WebSessions().Get(context.TODO(), types.GetWebSessionRequest{
+		User:      user,
+		SessionID: sessionID,
+	})
+}
+
+// UpsertWebSession updates or inserts a web session for the user and the session ID
+func (s *IdentityService) UpsertWebSession(user, sessionID string, session services.WebSession) error {
+	session.SetUser(user)
+	session.SetName(sessionID)
+	return s.WebSessions().Upsert(context.TODO(), session)
+}
+
+// DeleteWebSession deletes web session from the storage.
+func (s *IdentityService) DeleteWebSession(user, sessionID string) error {
+	return trace.Wrap(s.WebSessions().Delete(context.TODO(), types.DeleteWebSessionRequest{
+		User:      user,
+		SessionID: sessionID,
+	}))
+}
+
+// WebSessions returns the web sessions manager.
+func (s *IdentityService) WebSessions() services.WebSessionInterface {
+	return &webSessions{backend: s.Backend, log: s.log}
+}
+
+// Get returns the web session state described with req.
+func (r *webSessions) Get(ctx context.Context, req types.GetWebSessionRequest) (services.WebSession, error) {
+	if err := req.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	item, err := r.backend.Get(ctx, webSessionKey(req.SessionID))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	session, err := services.GetWebSessionMarshaler().UnmarshalWebSession(item.Value, services.SkipValidation())
```

Comment:
I'd just start importing from `types` instead of `services` in the new code to simplify future refactoring.